compiler/main.py

```diff
@@ -0,0 +1,75 @@
+import subprocess
+import os
+
+def execute_code(file_path):
+    # Define compile and run commands based on file extension
+    language_settings = {
+        '.py': {
+            'compile_cmd': None,
+            'run_cmd': 'python {file_path}'
+        },
+        '.cpp': {
+            'compile_cmd': 'g++ {file_path} -o {file_base}.out',
+            'run_cmd': '{file_base}.out'
+        },
+        '.java': {
+            'compile_cmd': 'javac {file_path}',
+            'run_cmd': 'java {file_base}'
+        },
+        '.js': {
+            'compile_cmd': None,
+            'run_cmd': 'node {file_path}'
+        },
+        '.rb': {
+            'compile_cmd': None,
+            'run_cmd': 'ruby {file_path}'
+        },
```

Comment:
can you convert the `run_cmd` to list, as it will have more
### Security: 
```
# Using a string with shell=True (vulnerable)
subprocess.run("ls; rm -rf /", shell=True)  # Dangerous if the input is not sanitized

# Using a list (safe)
subprocess.run(["ls", ";", "rm", "-rf", "/"])  # Will not execute as intended due to splitting
```
### Reliability:
```
# Using a string
subprocess.run('echo "Hello World!"', shell=True)  # Needs careful handling of quotes

# Using a list
subprocess.run(["echo", "Hello World!"])  # Arguments are passed directly
```
### Ease of Maintenance:
```
# Using a string
compile_cmd = 'g++ {} -o {}'.format(file_path, file_base + ".out")

# Using a list
compile_cmd = ["g++", file_path, "-o", file_base + ".out"]
```